src/nodejs/hooks/seeker_agent.go

```diff
@@ -0,0 +1,389 @@
+	seekerEnvironmentScript := "seeker-env.sh"
+
+	const seekerServerTemplate = "export SEEKER_SERVER_URL=%s\n"
+	scriptContent := fmt.Sprintf(seekerServerTemplate, h.serviceCredentials.SeekerServerURL)
+	stager.Logger().Info(seekerEnvironmentScript + " content: " + scriptContent)
+	return stager.WriteProfileD(seekerEnvironmentScript, scriptContent)
+}
+
+func extractServiceCredentials(Log *libbuildpack.Logger) (SeekerCredentials, error) {
+	type Service struct {
+		Name         string `json:"name"`
+		Label        string `json:"label"`
+		InstanceName string `json:"instance_name"`
+		BindingName  string `json:"binding_name"`
+		Credentials  struct {
+			EnterpriseServerUrl string `json:"enterprise_server_url"`
+			SeekerServerUrl     string `json:"seeker_server_url"`
+			SensorHost          string `json:"sensor_host"`
+			SensorPort          string `json:"sensor_port"`
+		} `json:"credentials"`
+	}
+
+	var vcapServices map[string][]Service
+
+	err := json.Unmarshal([]byte(os.Getenv("VCAP_SERVICES")), &vcapServices)
+	if err != nil {
+		return SeekerCredentials{}, fmt.Errorf("failed to unmarshal VCAP_SERVICES: %s", err)
+	}
+
+	var detectedCredentials []SeekerCredentials
+
+	for _, services := range vcapServices {
+		for _, service := range services {
+			if isSeekerRelated(service.Name, service.Label, service.InstanceName) {
+				credentials := SeekerCredentials{SeekerServerURL: service.Credentials.SeekerServerUrl}
+				detectedCredentials = append(detectedCredentials, credentials)
+			}
+		}
+	}
+
+	found, err := assertZeroOrOneServicesExist(len(detectedCredentials))
+	if err != nil {
+		return SeekerCredentials{}, err
+	}
+	if found {
+		return detectedCredentials[0], nil
+	}
+	return SeekerCredentials{}, nil
+}
+
+func assertZeroOrOneServicesExist(c int) (bool, error) {
+	if c > 1 {
+		return false, fmt.Errorf("expected to find 1 Seeker service but found %d", c)
+	}
+	if c == 1 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func extractServiceCredentialsUserProvidedService(Log *libbuildpack.Logger) (SeekerCredentials, error) {
+	type UserProvidedService struct {
+		BindingName interface{} `json:"binding_name"`
+		Credentials struct {
+			SeekerServerUrl string `json:"seeker_server_url"`
+		} `json:"credentials"`
+		InstanceName   string   `json:"instance_name"`
+		Label          string   `json:"label"`
+		Name           string   `json:"name"`
+		SyslogDrainURL string   `json:"syslog_drain_url"`
+		Tags           []string `json:"tags"`
+	}
+
+	type VCAPSERVICES struct {
+		UserProvidedService []UserProvidedService `json:"user-provided"`
+	} //`json:"VCAP_SERVICES"`
+
+	var vcapServices VCAPSERVICES
+	vcapServicesString := os.Getenv("VCAP_SERVICES")
+	if !strings.Contains(vcapServicesString, "user-provided") {
+		return SeekerCredentials{}, nil
+	}
+	err := json.Unmarshal([]byte(vcapServicesString), &vcapServices)
+	if err != nil {
+		return SeekerCredentials{}, fmt.Errorf("failed to unmarshal VCAP_SERVICES: %s", err.Error())
```

Comment:
Looks like we can just do something like the following and get all this logic to look a bit more simple:

```
err := json.Unmarshal([]byte(vcapServicesString), &vcapServices)
if err != nil {
  ...
}
if len(vcapServices.UserProvidedServices) == 0 {
  return SeekerCredentials{}, nil
}

Reply:
fixed